lib/vrl/stdlib/src/encode_logfmt.rs

```diff
@@ -0,0 +1,234 @@
+use vrl::prelude::*;
+
+mod logfmt {
+    use std::collections::{BTreeMap, HashSet};
+    use std::fmt::{self, Write};
+    use std::result::Result;
+
+    use vrl::prelude::*;
+
+    fn encode_string(output: &mut String, str: &str) -> fmt::Result {
+        let needs_quotting = str.find(' ').is_some();
+        if needs_quotting {
+            output.write_char('"')?;
+        }
+
+        for c in str.chars() {
```

Comment:
I am wondering if we should also be escaping things like newlines or carriage returns too. There was some related discussion to this in https://github.com/timberio/vector/pull/6739